aws/request.go

```diff
@@ -24,6 +24,47 @@ const (
 	ErrCodeRead = "ReadError"
 )
 
+// HTTPSerializationError provides a generic request serialization error
+type HTTPSerializationError struct {
+	Reason string // provides more context of error caused
+	Err    error  // original error
+}
+
+// Error returns a formatted error for HTTPSerializationError
+func (e *HTTPSerializationError) Error() string {
+	return fmt.Sprintf("serialization failed: %v %v", e.Reason, e.Err)
+}
+
+// Unwrap returns the underlying Error in HTTPDeserializationError
+func (e *HTTPSerializationError) Unwrap() error { return e.Err }
```

Comment:
Since `HTTPSerializationError` is generic, regardless of transport should this just be `SerializationError`?